### PR TITLE
Documentation and Cleanup

### PR DESCRIPTION
**Pull request by Kev-Y-Huang**
Generalized netlify functions and added some documentation for the functions

### CODE REVIEW

File: web/src/interactives/covid-dashboard/data/dataservice.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import axios from "axios";
 // api_url = 'https://documenter.getpostman.com/view/11144369/Szf6Z9B3?version=latest'
 
 const jhu_url = "https://corona.lmao.ninja/v2/";
+const HARVARD_COVID_URL = "1pJUpky9QRHA2tcaNMgS3gjq4pjrlUnjMZ2stXsaMOB0";
```
Comment (mwjkwan) [inline web/src/interactives/covid-dashboard/data/dataservice.js:6]: Let's create a folder called `constants` and store spreadsheet URLs there.

File: web/functions/cors/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# CORS Function
+
+Netlify function to set up CORS for API requests.
+
+## Usage
+
+First, the spreadsheet must be shared with the service account `hodp-netlify@hodp-sheets.iam.gserviceaccount.com`. Then, to use in code:
```
Comment (mwjkwan) [inline web/functions/cors/README.md:7]: Clarify whether the spreadsheet needs to be public, or whether it only needs to be shared with `hodp-netlify@hodp-sheets.iam.gserviceaccount.com`. Also, just made a [folder](https://drive.google.com/drive/u/0/folders/1ByD7GreoggyWpL-gpEgVRymVR_83XlcH) to consolidate our Google Spreadsheet data.

File: web/functions/algolia-search/algolia-search.js [hunk 3]
```diff
@@ -62,6 +64,7 @@ exports.handler = function (event, context, cb) {
               downloadURL: doc.downloadURL,
               sourceURL: doc.sourceURL,
             };
+          // Transforming article datatypes
```
Comment (mwjkwan) [inline web/functions/algolia-search/algolia-search.js:67]: Let's add blog posts to line 22 (which should share the properties with projects you specified below).

File: web/functions/google-spreadsheet/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Google Spreadsheet Function
+
+Netlify function to access Google Spreadsheets API fetching data stored in google spreadsheets.
```
Comment (mwjkwan) [inline web/functions/google-spreadsheet/README.md:3]: Link [this article](https://www.swyx.io/netlify-google-sheets/) as a resource

File: web/functions/google-spreadsheet/google-spreadsheet.js [hunk 5]
```diff
@@ -31,7 +28,7 @@ const { GoogleSpreadsheet } = require("google-spreadsheet");
 exports.handler = async (event) => {
   const table = event.queryStringParameters.table;
```
Comment (mwjkwan) [inline web/functions/google-spreadsheet/google-spreadsheet.js:29]: Super nitpicky, but I would unpack `event` here, like `const { headers, path, queryStringParameters } = event`.

File: web/functions/google-spreadsheet/google-spreadsheet.js [hunk 6]
```diff
@@ -11,17 +11,14 @@ if (!process.env.GOOGLE_SERVICE_ACCOUNT_EMAIL)
...
 /*
  * ok real work
  *
- * GET /.netlify/functions/google-spreadsheet-fn
- * GET /.netlify/functions/google-spreadsheet-fn/1
- * PUT /.netlify/functions/google-spreadsheet-fn/1
- * POST /.netlify/functions/google-spreadsheet-fn
- * DELETE /.netlify/functions/google-spreadsheet-fn/1
+ * GET /.netlify/functions/google-spreadsheet
+ * GET /.netlify/functions/google-spreadsheet/1
```
Comment (mwjkwan) [inline web/functions/google-spreadsheet/google-spreadsheet.js:18]: Replace `1` with `{rowNumber}` to be extra clear. Also, you can remove the "ok real work above" haha.

File: web/src/interactives/covid-dashboard/data/dataservice.js [hunk 7]
```diff
@@ -170,7 +172,7 @@ export const fetchHarvardData = async (table) => {
       tests: []
     };
 
-    const info = await fetch(`/.netlify/functions/harvard-covid?table=${table}`, {headers: {accept: "Accept: application/json"}});
+    const info = await fetch(`/.netlify/functions/google-spreadsheet?id=${HARVARD_COVID_URL}&table=${table}`, {headers: {accept: "Accept: application/json"}});
```
Comment (mwjkwan) [inline web/src/interactives/covid-dashboard/data/dataservice.js:175]: Nice

File: README.md [hunk 8]
```diff
@@ -1,5 +1,7 @@
 # [Harvard Open Data Project](https://www.hodp.org/)
 
+[![Netlify Status](https://api.netlify.com/api/v1/badges/181e9e5b-99d9-4e00-a1c4-da13d171b3a5/deploy-status)](https://app.netlify.com/sites/hodp/deploys)
```
Comment (mwjkwan) [inline README.md:3]: 🤩

**Review by mwjkwan — APPROVED** [review]
Looks really good! I left a few requests for clarification (so it's easier to understand for someone new to the tech team). Feel free to merge after you address those.

File: web/functions/algolia-search/algolia-search.js (hunk 3 above)
Comment (Kev-Y-Huang) [reply web/functions/algolia-search/algolia-search.js:67]: If they have the same properties, I'll just keep the switch cases the same and just make a note that the default takes both projects and posts into account

File: web/functions/cors/README.md (hunk 2 above)
Comment (Kev-Y-Huang) [reply web/functions/cors/README.md:7]: Great!

**Pull request merged**